src/tasks-prisma.service.ts

```diff
@@ -0,0 +1,62 @@
+import { Injectable } from '@nestjs/common';
+import { PrismaService } from './prisma.service';
+import { prisma, Prisma, task, user } from '@prisma/client';
+import { TaskStatus } from './tasks/task.status.enum';
+import { CreateTaskDto } from './tasks/dto/create-task.dto';
+import { GetTasksFilterDto } from './tasks/dto/get-tasks-filter.dto';
+
+@Injectable()
+export class TasksPrismaService {
+  constructor(private prisma: PrismaService) {}
+
+  async createTask(data: CreateTaskDto, user: user): Promise<task> {
+    const { title, description } = data;
+
+    const userTasks = await this.prisma.task.create({
+      data: {
+        title,
+        description,
+        status: TaskStatus.OPEN,
+        user: {
+          connect: { id: user.id },
+        },
+      },
+    });
+
+    return userTasks;
+  }
+
+  async getTasks(data: GetTasksFilterDto, user: user): Promise<task[]> {
+    const { status, search } = data;
+
+    const result = this.prisma.task.findMany({
+      where: {
+        AND: [
+          {
+            userId: user.id,
+          },
+          {
+            status: status,
+          },
+          {
+            OR: [
+              {
+                description: {
+                  contains: search,
+                  mode: 'insensitive',
+                },
+              },
+              {
+                title: {
+                  contains: search,
+                  mode: 'insensitive',
+                },
+              },
+            ],
+          },
+        ],
+      },
+    });
+    return result;
+  }
+}
```

Comment:
For this service, it may be better to choose a name other than `TasksPrismaService`, and to consider a different location within the codebase. As this service is intended to replace the old `TasksService`, it may be appropriate to remove the `TasksRepository` as well. Additionally, since we are transitioning away from typeORM, we will need to replace any related functionality.